### PR TITLE
Gm focus

### PR DESCRIPTION
**Pull request by mgalbier**
- Fix condition for dependsOn function
- Fix indentation
- Add border in ReviewComment visualization to allow for easier moving of comments
- Improve name change handling
- Add command to toggle between different levels of detail for name change highlights
- Rename nameChanges_ to nameChangeInformation_
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75594938%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75594999%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75595006%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75595053%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75595157%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75595186%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23issuecomment-241257244%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23issuecomment-241259148%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23issuecomment-241261170%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23issuecomment-241257244%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22OK%2C%20I%27m%20done%20with%20the%20review.%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A27%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Thanks%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A57%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%2C%20%7B%22body%22%3A%20%22Thanks.%22%2C%20%22created_at%22%3A%20%222016-08-21T14%3A35%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%209360be61cad6e93e24fe5a817d3ec64c50e9b02c%20VersionControlUI/src/DiffManager.h%2095%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75594999%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Uf%2C%20there%27s%20a%20ton%20of%20these%20statics%20related%20to%20name%20changes.%20Could%20you%20at%20least%20put%20some%20comments%2C%20to%20explain%20in%20a%20sentence%20or%20two%20what%20each%20one%20is%20for%3F%20Is%20it%20possible%20to%20perhaps%20combine%20some%20of%20these%20%28maybe%20by%20making%20an%20additional%20class/struct%29%3F%20Would%20such%20a%20combination%20be%20a%20better%20design%3F%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A13%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VersionControlUI/src/DiffManager.h%3AL204-234%22%7D%2C%20%22Pull%209360be61cad6e93e24fe5a817d3ec64c50e9b02c%20VersionControlUI/src/DiffManager.h%2099%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75595006%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20is%20this%20needed%3F%5Cr%5Cn%5Cr%5CnThere%27s%20really%20a%20lot%20of%20these%20things.%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A14%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20did%20it%20because%20the%20command%20%60toggleNameChangeHighlights%60%20is%20static.%20They%20contain%20the%20information%20of%20the%20last%20diff%20and%20I%20use%20this%20to%20toggle%20the%20highlights.%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A24%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VersionControlUI/src/DiffManager.h%3AL204-234%22%7D%2C%20%22Pull%209360be61cad6e93e24fe5a817d3ec64c50e9b02c%20VersionControlUI/src/DiffManager.cpp%2033%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75595053%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22please%2C%20use%20the%20same%20multiplicity%20for%20both%20the%20type%20name%20and%20the%20variable%20names.%20It%27s%20kind%20of%20strange%20for%20something%20of%20type%20%60Visualizations%60%20to%20be%20called%20%60visualization%60.%5Cr%5Cn%5Cr%5CnAlso%20this%20should%20be%20initialized%20above%2C%20like%20the%20rest%20of%20the%20fields%2C%20after%20the%20%60%3A%60.%5Cr%5Cn%5Cr%5CnAlso%2C%20I%20think%20the%20name%20should%20be%20a%20little%20more%20explanatory%2C%20include%20%60flags%60%20at%20the%20end.%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A17%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VersionControlUI/src/DiffManager.cpp%3AL57-79%22%7D%2C%20%22Pull%209360be61cad6e93e24fe5a817d3ec64c50e9b02c%20VersionControlUI/src/VersionControlUIPlugin.cpp%2017%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75594938%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20put%20this%20on%20top%20of%20the%20file.%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A09%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VersionControlUI/src/VersionControlUIPlugin.cpp%3AL24-44%22%7D%2C%20%22Pull%209360be61cad6e93e24fe5a817d3ec64c50e9b02c%20VersionControlUI/src/DiffManager.cpp%20361%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/439%23discussion_r75595186%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Wow%2C%20this%20name%20change%20stuff%20resulted%20in%20a%20ton%20more%20changes%20than%20I%20expected.%5Cr%5Cn%5Cr%5CnI%27m%20thinking%2C%20it%20might%20make%20sense%20to%20move%20all%20name%20change%20related%20things%20to%20a%20separate%20class.%20Let%27s%20talk%20about%20this%20in%20our%20next%20meeting.%20Think%20about%20it%20in%20the%20mean%20time.%22%2C%20%22created_at%22%3A%20%222016-08-21T13%3A27%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VersionControlUI/src/DiffManager.cpp%3AL856-886%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 9360be61cad6e93e24fe5a817d3ec64c50e9b02c VersionControlUI/src/VersionControlUIPlugin.cpp 17'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/439#discussion_r75594938'>File: VersionControlUI/src/VersionControlUIPlugin.cpp:L24-44</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Please put this on top of the file.
- [ ] <a href='#crh-comment-Pull 9360be61cad6e93e24fe5a817d3ec64c50e9b02c VersionControlUI/src/DiffManager.h 95'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/439#discussion_r75594999'>File: VersionControlUI/src/DiffManager.h:L204-234</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Uf, there's a ton of these statics related to name changes. Could you at least put some comments, to explain in a sentence or two what each one is for? Is it possible to perhaps combine some of these (maybe by making an additional class/struct)? Would such a combination be a better design?
- [ ] <a href='#crh-comment-Pull 9360be61cad6e93e24fe5a817d3ec64c50e9b02c VersionControlUI/src/DiffManager.h 99'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/439#discussion_r75595006'>File: VersionControlUI/src/DiffManager.h:L204-234</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Why is this needed?
  There's really a lot of these things.
- <a href='https://github.com/mgalbier'></a> I did it because the command `toggleNameChangeHighlights` is static. They contain the information of the last diff and I use this to toggle the highlights.
- [ ] <a href='#crh-comment-Pull 9360be61cad6e93e24fe5a817d3ec64c50e9b02c VersionControlUI/src/DiffManager.cpp 33'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/439#discussion_r75595053'>File: VersionControlUI/src/DiffManager.cpp:L57-79</a></b>
- <a href='https://github.com/dimitar-asenov'></a> please, use the same multiplicity for both the type name and the variable names. It's kind of strange for something of type `Visualizations` to be called `visualization`.
  Also this should be initialized above, like the rest of the fields, after the `:`.
  Also, I think the name should be a little more explanatory, include `flags` at the end.
- [ ] <a href='#crh-comment-Pull 9360be61cad6e93e24fe5a817d3ec64c50e9b02c VersionControlUI/src/DiffManager.cpp 361'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/439#discussion_r75595186'>File: VersionControlUI/src/DiffManager.cpp:L856-886</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Wow, this name change stuff resulted in a ton more changes than I expected.
  I'm thinking, it might make sense to move all name change related things to a separate class. Let's talk about this in our next meeting. Think about it in the mean time.
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/439#issuecomment-241257244'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'></a> OK, I'm done with the review.
- <a href='https://github.com/mgalbier'></a> Thanks
- <a href='https://github.com/dimitar-asenov'></a> Thanks.

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/439?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/439?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/439'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/439'>Refresh</a>.<a href='#crh-update-none'></a>

File: VersionControlUI/src/VersionControlUIPlugin.cpp [hunk 1]
```diff
@@ -24,18 +24,20 @@
...
-#include "InteractionBase/src/handlers/HSceneHandlerItem.h"
-
+#include "DiffManager.h"
 #include "items/ObjectPathCrumb.h"
 #include "handlers/HObjectPathCrumb.h"
-
 #include "commands/CClear.h"
 #include "commands/CDiff.h"
 #include "commands/CHistory.h"
+#include "VersionControlUIPlugin.h"
```
Comment (dimitar-asenov) [inline VersionControlUI/src/VersionControlUIPlugin.cpp]: Please put this on top of the file.

File: VersionControlUI/src/DiffManager.h [hunk 2]
```diff
@@ -181,23 +204,30 @@ class VERSIONCONTROLUI_API DiffManager
...
 		QList<Model::SymbolMatcher> contextUnitMatcherPriorityList_;
 
 		// if set specifies which node we are interested in, used for history
 		Model::NodeIdType targetNodeId_;
 
-		QHash<QString, QPair<QString, Model::NodeIdType>> nameChanges_;
-		QHash<Model::NodeIdType, bool> nameChangesIdsIsNameText_;
-		NameChangeVisualizations nameChangeVisualization_{Summary};
+		QHash<QString, QPair<QString, Model::NodeIdType>> nameChangeInformation_;
+		static QHash<Model::NodeIdType, bool> nameChangesIdsIsNameText_;
```
Comment (dimitar-asenov) [inline VersionControlUI/src/DiffManager.h:223]: Uf, there's a ton of these statics related to name changes. Could you at least put some comments, to explain in a sentence or two what each one is for? Is it possible to perhaps combine some of these (maybe by making an additional class/struct)? Would such a combination be a better design?

File: VersionControlUI/src/DiffManager.h [hunk 3]
```diff
@@ -181,23 +204,30 @@ class VERSIONCONTROLUI_API DiffManager
...
+		static QList<ChangeWithNodes> nameChanges_;
+		static NameChangeVisualizations nameChangeVisualization_;
+		static QList<int> nameChangeOnZoomHandlerIds_;
+		static QSet<Visualization::Item*> nameChangesScaledByAncestor_;
```
Comment (dimitar-asenov) [inline VersionControlUI/src/DiffManager.h:244]: Why is this needed?

There's really a lot of these things.

File: VersionControlUI/src/DiffManager.cpp [hunk 4]
```diff
@@ -57,25 +57,22 @@ namespace VersionControlUI
...
 
 DiffManager::DiffManager(QString project, QList<Model::SymbolMatcher> contextUnitMatcherPriorityList,
 								 Model::NodeIdType targetNodeID, NameChangeVisualizations nameChangeVisualization)
-	: project_{project}, contextUnitMatcherPriorityList_{contextUnitMatcherPriorityList}, targetNodeId_{targetNodeID},
-	  nameChangeVisualization_{nameChangeVisualization}
-{}
+	: project_{project}, contextUnitMatcherPriorityList_{contextUnitMatcherPriorityList}, targetNodeId_{targetNodeID}
+
+{
+	nameChangeVisualization_ = nameChangeVisualization;
```
Comment (dimitar-asenov) [inline VersionControlUI/src/DiffManager.cpp]: please, use the same multiplicity for both the type name and the variable names. It's kind of strange for something of type `Visualizations` to be called `visualization`.

Also this should be initialized above, like the rest of the fields, after the `:`.

Also, I think the name should be a little more explanatory, include `flags` at the end.

File: VersionControlUI/src/DiffManager.h (hunk 3 above)
Comment (mgalbier) [inline VersionControlUI/src/DiffManager.h:244]: I did it because the command `toggleNameChangeHighlights` is static. They contain the information of the last diff and I use this to toggle the highlights.

File: VersionControlUI/src/DiffManager.cpp [hunk 5]
```diff
@@ -766,13 +856,30 @@ void DiffManager::createOverlaysForChanges(Visualization::ViewItem* diffViewItem
...
+
+	}
+
+	if (!nameChangeItemsToScale.isEmpty())
+	{
+		nameChangeItemsToScale.subtract(nameChangesScaledByAncestor_);
+		scaleItems(nameChangeItemsToScale, diffViewItem, true);
+	}
+
+	diffViewItem->setUpdateNeeded(Visualization::Item::StandardUpdate);
```
Comment (dimitar-asenov) [inline VersionControlUI/src/DiffManager.cpp:880]: Wow, this name change stuff resulted in a ton more changes than I expected.

I'm thinking, it might make sense to move all name change related things to a separate class. Let's talk about this in our next meeting. Think about it in the mean time.

**Comment by dimitar-asenov** [comment]
OK, I'm done with the review.

**Comment by mgalbier** [comment]
Thanks

**Comment by dimitar-asenov** [comment]
Thanks.
